js/client-chat.js

```diff
@@ -186,6 +186,8 @@
 
 		getHighlight: function (message) {
 			var highlights = Tools.prefs('highlights') || [];
+			var roomHighlights = (this.type === 'battle' ? Tools.prefs('battlehls') || [] : Tools.prefs(this.id) || []);
```

Comment:
no, you probably want it as data in `'highlights'`. You shouldn't take up more than one prefs entry for any one feature.